src/webview/apps/lightspeed/roleGeneration/main.ts

```diff
@@ -13,6 +12,53 @@ import {
 } from "@vscode/webview-ui-toolkit";
 import { EditableList } from "../../common/editableList";
 
+import { getSingletonHighlighterCore, HighlighterCore } from "@shikijs/core";
+import darkPlus from "shiki/themes/dark-plus.mjs";
+import lightPlus from "shiki/themes/light-plus.mjs";
+import { default as yamlLang } from "shiki/langs/yaml.mjs";
+import getWasm from "shiki/wasm";
+
+let highlighter: HighlighterCore;
```

Comment:
Something in my mind is telling me this should be a member of the `codeToHtml(..)` function.

i.e.
```
export async function codeToHtml(code: string) {

  let highlighter: HighlighterCore;

  function is DarkMode(): boolean {
    ...
```

IDK what makes me think that. It can probably be ignored...

Reply:
The problem is that it would need access to the DOM.